src/reporters.ts

```diff
@@ -375,3 +377,60 @@ export const escape = (value: string): string => {
   }
   return value;
 };
+
+export const BatchInfoColumns = {
+  id: { header: 'Batch Id' },
+  state: { header: 'State' },
+  failed: { header: 'Failed' },
+  stateMessage: { header: 'Message' },
+};
+
+export const getBatchTotals = (batches: BatchInfo[]): { total: number; failed: number; success: number } => {
+  const ttls = batches.reduce(
+    (acc: { total: number; failed: number; success: number }, batch) => {
+      acc.total += parseInt(batch.numberRecordsProcessed, 10);
+      acc.failed += parseInt(batch.numberRecordsFailed, 10);
+      acc.success = acc.total - acc.failed;
+      return acc;
+    },
+    { total: 0, failed: 0, success: 0 }
+  );
+  return ttls;
+};
+
+export function getBulk2JobTotals<J extends Schema>(
+  results: IngestJobV2Results<J>
+): { total: number; failed: number; success: number; unprocessed: number } {
+  const ttls = { total: 0, failed: 0, success: 0, unprocessed: 0 };
+  ttls.total =
+    Object.keys(results.successfulResults).length +
+    Object.keys(results.failedResults).length +
+    Object.keys(results.unprocessedRecords).length;
+  ttls.failed = Object.keys(results.failedResults).length;
+  ttls.success = Object.keys(results.successfulResults).length;
+  ttls.unprocessed = Object.keys(results.unprocessedRecords).length;
+  return ttls;
+}
+
+export const getFailedBatchesForDisplay = (
+  batches: BatchInfo[]
+): Array<{ id: string; state: BatchState; failed: string; stateMessage: string }> => {
+  const failedBatches = batches.filter((batch) => parseInt(batch.numberRecordsFailed, 10) > 0);
```

Comment:
can you combine this filter/parseInt with the map below, and just return that without creating a new constant?